test/FunctionalTests/Microsoft.OData.Core.Tests/ScenarioTests/Roundtrip/JsonReaderWriterInjectionTests.cs

```diff
@@ -125,8 +125,13 @@ private static void RunReaderTest(Action<IContainerBuilder> action, string messa
             var entitySet = model.FindDeclaredEntitySet("People");
             var entityType = model.GetEntityType("NS.Person");
             var container = ContainerBuilderHelper.BuildContainer(action);
-            container.GetRequiredService<ODataSimplifiedOptions>().EnableReadingODataAnnotationWithoutPrefix = true;
-            var resource = GetReadedResource(messageContent, model, entitySet, entityType, container);
+
+            var readerSettings = new ODataMessageReaderSettings
+            {
+                EnableReadingODataAnnotationWithoutPrefix = true
+            };
+
+            var resource = GetReadedResource(messageContent, model, entitySet, entityType, container, readerSettings);

```

Comment:
Why not inject the `ODataMessageReaderSettings` into the DI container so it's retrieved from `GetReaderResource` method and you don't need to change the method signature?

Reply:
Because the test was instantiating a new `ODataMessageWriterSettings` and passing it directly to the `ODataMessageReader` without using DI. So passing it directly felt like a more direct port of the previous behavior.